coretex/cli/modules/utils.py

```diff
@@ -15,22 +15,64 @@
 #     You should have received a copy of the GNU Affero General Public License
 #     along with this program.  If not, see <https://www.gnu.org/licenses/>.
 
-from typing import List, Any, Optional, Callable
+from typing import List, Any, Tuple, Optional, Callable
 from functools import wraps
 
 import sys
+import venv
 
 from py3nvml import py3nvml
 
 import click
+import platform
 
+from ...configuration import DEFAULT_VENV_PATH
 from ...utils.process import command
 
 
+def fetchCoretexVersion() -> Optional[str]:
+    _, output, _ = command(["pip", "freeze"], ignoreStdout = True)
+    packages = output.splitlines()
+
+    for package in packages:
+        if "coretex" in package:
+            return package.replace(" ", "")
+
+    return None
+
+
+def checkEnvironment() -> None:
+    venvPython = DEFAULT_VENV_PATH / "bin" / "python"
+    if DEFAULT_VENV_PATH.exists():
+        return
+
+    venv.create(DEFAULT_VENV_PATH, with_pip = True)
+
+    if platform.system() == "Windows":
+        venvPython = DEFAULT_VENV_PATH / "Scripts" / "python.exe"
+
+    currentCtxVersion = fetchCoretexVersion()
+    if currentCtxVersion is not None:
+        command([str(venvPython), "-m", "pip", "install", currentCtxVersion], ignoreStdout = True)
+
+
 def updateLib() -> None:
     command([sys.executable, "-m", "pip", "install", "--no-cache-dir", "--upgrade", "coretex"], ignoreStdout = True, ignoreStderr = True)
 
 
+def getExecPaths() -> Tuple[str, str]:
+    _, dockerPath, _ = command(["which", "docker"], ignoreStdout = True, ignoreStderr = True)
+    _, gitPath, _ = command(["which", "git"], ignoreStdout = True, ignoreStderr = True)
+
+    dockerPathParts = dockerPath.strip().split('/')
+    dockerExecPath = '/'.join(dockerPathParts[:-1])
+
+    gitPathParts = gitPath.strip().split('/')
+    gitExecPath = '/'.join(gitPathParts[:-1])
+
+    return dockerExecPath, gitExecPath
```

Comment:
This should be a function which gets the path to a binary of a single executable. ex. `getExecPath("docker")`, `getExecPath("git")`, but not for both at the same time